tests/lib/mkpinentry.sh

```diff
@@ -1,21 +1,30 @@
 #!/bin/bash
 
+prepare_pinentry_v2(){
+    mkdir -p ~/.snap/gnupg/
+    # It is just created once by gpg-agent, then if this dir does not exist gpg will
+    # fail to create the key
+    mkdir -p ~/.snap/gnupg/private-keys-v1.d
+    echo pinentry-program "$TESTSLIB/pinentry-fake.sh" > ~/.snap/gnupg/gpg-agent.conf
+    chmod -R go-rwx ~/.snap
+}
+
 if gpg --version | MATCH 'gpg \(GnuPG\) 1.'; then
     echo "fake gpg pinentry not used for gpg v1"
 else
     echo "setup fake gpg pinentry environment for gpg v2 or higher"
     case "$SPREAD_SYSTEM" in
         opensuse-*)
-            mkdir -p ~/.gnupg
-            echo pinentry-program "$TESTSLIB/pinentry-fake.sh" > ~/.gnupg/gpg-agent.conf
+            if gpg --version | MATCH 'gpg \(GnuPG\) 2.0'; then
+                mkdir -p ~/.gnupg
+                echo "no-grab" > ~/.gnupg/gpg-agent.conf
```

Comment:
I assume that `chmod -R go-rwx` and `no-grab` are not really specific to v2.0 (or v2.1+), but rather `~/.snap/gnupg` vs `~/.gnupg` is. In that case, maybe we could unify this a bit:
```
if gpg --version | MATCH 'gpg \(GnuPG\) 1.'; then
    echo "fake gpg pinentry not used for gpg v1"
else
    echo "setup fake gpg pinentry environment for gpg v2 or higher"
    gpgdir=~/.gnupg
    if gpg --version | MATCH 'gpg \(GnuPG\) 2.0'; then
        gpgdir=~/.snap/gnupg
    fi
    mkdir -p "$gpgdir"
    # It is just created once by gpg-agent, then if this dir does not exist gpg will
    # fail to create the key
    mkdir -p "$gpgdir/private-keys-v1.d"
    echo "no-grab" > "$gpgdir/gpg-agent.conf"
    echo pinentry-program "$TESTSLIB/pinentry-fake.sh" > "$gpgdir/gpg-agent.conf"
    chmod -R go-rwx "$gpgdir"
    test -d ~/.snap && chmod -R go-rwx ~/.snap
fi
```

Reply:
Great, I'll try this.